### PR TITLE
Debug Info v2

### PR DESCRIPTION
**Pull request by devhawk**
This is a draft PR for the updated Debug Info used by the [Neo Debugger Storage Schema Preview](https://github.com/neo-project/neo-debugger/blob/master/docs/storage-schema-overview.md).

Changes are not expected, but until the storage schema work is out of preview, this PR will stay in draft status.

### CODE REVIEW

File: nep-19.mediawiki [hunk 1]
```diff
@@ -22,11 +24,248 @@ about parameters, variables and storage items that exists in the contract source
...
+
+This section describes both the abstract ContractType model as well as the concrete syntax used to 
+string encoding ContractTypes.
+
+Note, because the ContractType model namespace is unbounded (i.e. contract developers can define
+their own named structs), built-in ContractTypes use a '''#''' prefix when serialized. This eliminates
+the possibility of a name collision since '''#''' is not a supported identifier character in most
+programming languages.
+
+The Version 2 type model includes six concrete ContractTypes.
```
Comment (devhawk) [inline nep-19.mediawiki:100]: Need to add a mechanism to encode Iterators - probably as an intrinsic type (i.e. #Iterator).

**Comment by ixje** [comment]
What is it we are waiting for for this to be merged?

**Comment by roman-khimov** [comment]
Can't tell anything about the "draft" status of it (maybe there are some changes planned), but I'd say it should be a new NEP replacing the old one. That's exactly why we have "Replaces" and "Superseded-By" in NEPs. And then compilers could be compatible with NEP-19 or with new NEP-XYZ.

**Comment by ixje** [comment]
apparently it's waiting to be updated after https://github.com/neo-project/proposals/pull/160 is accepted

**Pull request closed without merging**